### PR TITLE
feat: Add labels to setup Pods for easy logs collection

### PR DESCRIPTION
**Pull request by harshshahsumo**
[OSC-832] Add labels to setup Pods for easy logs collection

[OSC-832]: https://sumologic.atlassian.net/browse/OSC-832?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: deploy/helm/sumologic/templates/_helpers/_common.tpl [hunk 1]
```diff
@@ -149,6 +151,7 @@ machineconfiguration.openshift.io/role: worker
 
 {{- define "sumologic.labels.app.machineconfig.master" -}}
 {{- template "sumologic.fullname" . }}-master-extensions
+sumologic.com/app: otelcol-events
```
Comment (harshshahsumo) [inline deploy/helm/sumologic/templates/_helpers/_common.tpl:154]: not sure if this is the right place?

File: deploy/helm/sumologic/templates/setup/job.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ metadata:
 {{ include "sumologic.annotations.app.setup.helmsh" "3" | indent 4 }}
   labels:
     app: {{ template "sumologic.labels.app.setup.job" . }}
+    sumologic.com/app: setup
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/setup/job.yaml:17]: I would creat new helm template to have both of the labels

File: deploy/helm/sumologic/templates/_helpers/_common.tpl (hunk 1 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/templates/_helpers/_common.tpl:154]: no it's not. You should create new template analogically, I would go with `sumologic.labels.app.events` name

File: deploy/helm/sumologic/templates/_helpers/_common.tpl [hunk 3]
```diff
@@ -155,7 +155,11 @@ machineconfiguration.openshift.io/role: worker
 machineconfiguration.openshift.io/role: master
 {{- end -}}
 
-{{/*
+{{- define "sumologic.labels.otelcol.events" -}}
```
Comment (chan-tim-sumo) [inline deploy/helm/sumologic/templates/_helpers/_common.tpl:158]: hmm this is in a common file, should this be in a new file instead?

File: deploy/helm/sumologic/templates/setup/job.yaml [hunk 4]
```diff
@@ -15,6 +15,7 @@ metadata:
   labels:
     app: {{ template "sumologic.labels.app.setup.job" . }}
     {{- include "sumologic.labels.common" . | nindent 4 }}
+    {{- include "sumologic.labels.otelcol.events" . | nindent 8 }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/setup/job.yaml:18]:
```suggestion
    {{- include "sumologic.labels.otelcol.events" . | nindent 4 }}
```

`events` shouldn't be in `setup`

File: deploy/helm/sumologic/templates/setup/job.yaml [hunk 5]
```diff
@@ -28,6 +29,7 @@ spec:
 {{- end }}
       labels:
 {{- if .Values.sumologic.podLabels }}
+{{- include "sumologic.labels.otelcol.events" . | nindent 8 }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/setup/job.yaml:32]:
```suggestion
{{- include "sumologic.labels.otelcol.events" . | nindent 8 }}
{{- if .Values.sumologic.podLabels }}
```
this include should be outside of `if` and also `events` shouldn't be in `setup`

**Review by sumo-drosiek — COMMENTED** [review]
```
➜  sumologic-kubernetes-collection git:(osc832_AddLabelsToSetupPods) helm template deploy/helm/sumologic --set sumologic.accessId=dummy --set sumologic.accessKey=dummy
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/drosiek/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/drosiek/.kube/config
Error: YAML parse error on sumologic/templates/setup/job.yaml: error converting YAML to JSON: yaml: line 14: did not find expected key

Use --debug flag to render out invalid YAML
```

https://github.com/SumoLogic/sumologic-kubernetes-collection/actions/runs/10187187667/job/28180686924?pr=3817

>  Error:  templates/setup/job.yaml: unable to parse YAML: error converting YAML to JSON: yaml: line 18: did not find expected key

There is a problem with generating `templates/setup/job.yaml`

You can generate the file locally with the following command:
`helm template deploy/helm/sumologic --set sumologic.accessId=dummy --set sumologic.accessKey=dummy -s templates/setup/job.yaml --debug`

```
➜  sumologic-kubernetes-collection git:(osc832_AddLabelsToSetupPods) helm template deploy/helm/sumologic --set sumologic.accessId=dummy --set sumologic.accessKey=dummy -s templates/setup/job.yaml --debug
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/drosiek/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/drosiek/.kube/config
install.go:222: [debug] Original chart version: ""
install.go:239: [debug] CHART PATH: /Users/drosiek/projects/sumologic-kubernetes-collection/deploy/helm/sumologic

---
# Source: sumologic/templates/setup/job.yaml




apiVersion: batch/v1
kind: Job
metadata:
  name: release-name-sumologic-setup
  namespace: sumologic
  annotations:
    helm.sh/hook: pre-install,pre-upgrade
    helm.sh/hook-weight: "3"
    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
  labels:
    app: release-name-sumologic
    chart: "sumologic-4.9.0"
    release: "release-name"
    heritage: "Helm"
        sumologic.com/app: otelcol-events
spec:
  ttlSecondsAfterFinished: 120
  template:
    metadata:
      annotations:
      labels:
    spec:
      restartPolicy: OnFailure
      serviceAccountName: release-name-sumologic-setup
      nodeSelector:
        kubernetes.io/os: linux
      volumes:
      - name: setup
        configMap:
          name: release-name-sumologic-setup
          defaultMode: 0777
      - name: custom
        configMap:
          name: release-name-sumologic-setup-custom
          defaultMode: 0777
      containers:
      - name: setup
        image: public.ecr.aws/sumologic/kubernetes-setup:3.15.1
        imagePullPolicy: IfNotPresent
        command: ["/etc/terraform/setup.sh"]
        resources:
          limits:
            cpu: 2000m
            memory: 256Mi
          requests:
            cpu: 200m
            memory: 64Mi
        volumeMounts:
        - name: setup
          mountPath: /etc/terraform
        - name: custom
          mountPath: /customer-scripts
        envFrom:
        - secretRef:
            name: release-name-sumologic-setup
        env:
        - name: NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: SUMOLOGIC_BASE_URL
          value: 
        - name: SUMOLOGIC_COLLECTOR_NAME
          value: kubernetes
        - name: SUMOLOGIC_SECRET_NAME
          value: "sumologic"
        - name: CHART_VERSION
          value: "4.9.0"
        - name: SUMOLOGIC_MONITORS_ENABLED
          value: "true"
        - name: SUMOLOGIC_MONITORS_STATUS
          value: "enabled"
        - name: SUMOLOGIC_DASHBOARDS_ENABLED
          value: "true"
        
        - name: NO_PROXY
          value: kubernetes.default.svc
      securityContext:
        runAsUser: 1000
Error: YAML parse error on sumologic/templates/setup/job.yaml: error converting YAML to JSON: yaml: line 14: did not find expected key
helm.go:84: [debug] error converting YAML to JSON: yaml: line 14: did not find expected key
YAML parse error on sumologic/templates/setup/job.yaml
helm.sh/helm/v3/pkg/releaseutil.(*manifestFile).sort
	helm.sh/helm/v3/pkg/releaseutil/manifest_sorter.go:146
helm.sh/helm/v3/pkg/releaseutil.SortManifests
	helm.sh/helm/v3/pkg/releaseutil/manifest_sorter.go:106
helm.sh/helm/v3/pkg/action.(*Configuration).renderResources
	helm.sh/helm/v3/pkg/action/action.go:168
helm.sh/helm/v3/pkg/action.(*Install).RunWithContext
	helm.sh/helm/v3/pkg/action/install.go:312
main.runInstall
	helm.sh/helm/v3/cmd/helm/install.go:314
main.newTemplateCmd.func2
	helm.sh/helm/v3/cmd/helm/template.go:95
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.8.0/command.go:983
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.8.0/command.go:1115
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.8.0/command.go:1039
main.main
	helm.sh/helm/v3/cmd/helm/helm.go:83
runtime.main
	runtime/proc.go:271
runtime.goexit
	runtime/asm_arm64.s:1222
```

And then, we can see that the indentation around 18th line is broken:
```
labels:
    app: release-name-sumologic
    chart: "sumologic-4.9.0"
    release: "release-name"
    heritage: "Helm"
        sumologic.com/app: otelcol-events
```

it should be

```
labels:
    app: release-name-sumologic
    chart: "sumologic-4.9.0"
    release: "release-name"
    heritage: "Helm"
    sumologic.com/app: otelcol-events
```

**Review by sumo-drosiek — CHANGES_REQUESTED** [review]
Please do not add `sumologic.com/app: otelcol-events` label to setup pods. `Setup` is not `events` and it's not `otelcol`

File: deploy/helm/sumologic/templates/_helpers/_common.tpl [hunk 6]
```diff
@@ -77,6 +77,7 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 
 {{- define "sumologic.labels.app.setup" -}}
 {{- template "sumologic.labels.app" . }}
+sumologic.com/app: otelcol-events
```
[the remark this answers is not in the document]
Comment (rnishtala-sumo) [reply deploy/helm/sumologic/templates/_helpers/_common.tpl:83]: shouldn't this be `sumologic.com/app: setup`?  as stated in the [github issue](https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/3801) The label value of `otelcol-events` doesn't seem to be a correct identifier for the setup pod.

File: deploy/helm/sumologic/templates/_helpers/_common.tpl [hunk 7]
```diff
@@ -155,7 +159,7 @@ machineconfiguration.openshift.io/role: worker
 machineconfiguration.openshift.io/role: master
 {{- end -}}
 
-{{/*
+{{/*
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/_helpers/_common.tpl:162]:
```suggestion
{{/*
```

File: deploy/helm/sumologic/templates/setup/job.yaml (hunk 5 above)
Comment (harshshahsumo) [reply deploy/helm/sumologic/templates/setup/job.yaml]: fixed

File: deploy/helm/sumologic/templates/_helpers/_common.tpl (hunk 7 above)
Comment (harshshahsumo) [reply deploy/helm/sumologic/templates/_helpers/_common.tpl]: addressed

File: deploy/helm/sumologic/templates/setup/job.yaml (hunk 2 above)
Comment (harshshahsumo) [reply deploy/helm/sumologic/templates/setup/job.yaml]: done

**Review by sumo-drosiek — APPROVED** [review]
LGTM, addressing @chan-tim-sumo comment, We don't have separate file for setup right now. If we feel, we need it, we can move setup related templates to new file in another PR

**Pull request merged**